editor/store/effects.js

```diff
@@ -216,14 +216,6 @@ export default {
 				{ id: SAVE_POST_NOTICE_ID, spokenMessage: noticeMessage }
 			) );
 		}
-
-		if ( get( window.history.state, [ 'id' ] ) !== post.id ) {
-			window.history.replaceState(
-				{ id: post.id },
-				'Post ' + post.id,
-				getPostEditUrl( post.id )
-			);
-		}
```

Comment:
I love that this moves the URL change to a "side-effect" outside the `editor` module. Nice work